python/ray/tune/tune.py

```diff
@@ -4,28 +4,49 @@
 
 import time
 
-from ray.tune import TuneError
-from ray.tune.hyperband import HyperBandScheduler
-from ray.tune.async_hyperband import AsyncHyperBandScheduler
-from ray.tune.median_stopping_rule import MedianStoppingRule
+from ray.tune.trial_scheduler import FIFOScheduler
+from ray.tune.error import TuneError
 from ray.tune.trial import Trial, DEBUG_PRINT_INTERVAL
 from ray.tune.log_sync import wait_for_log_sync
 from ray.tune.trial_runner import TrialRunner
-from ray.tune.trial_scheduler import FIFOScheduler
 from ray.tune.web_server import TuneServer
-from ray.tune.variant_generator import generate_trials
-from ray.tune.experiment import Experiment
+from ray.tune.experiment import Experiment, JSONExperiment
+
+
+_SCHEDULERS = {}
+
+
+def get_scheduler(alg):
+    if alg == "FIFO":
+        return FIFOScheduler
+    elif alg == "MedianStopping":
+        from ray.tune.median_stopping_rule import MedianStoppingRule
+        return MedianStoppingRule
+    elif alg == "HyperBand":
+        from ray.tune.hyperband import HyperBandScheduler
+        return HyperBandScheduler
+    elif alg == "AsyncHyperBand":
+        from ray.tune.async_hyperband import AsyncHyperBandScheduler
+        return AsyncHyperBandScheduler
+    elif alg == "HyperOpt":
+        from ray.tune.hpo_scheduler import HyperOptScheduler
+        return HyperOptScheduler
+    else:
+        raise Exception(
+            ("Unknown scheduler {}.").format(alg))
 
 
-_SCHEDULERS = {
-    "FIFO": FIFOScheduler,
-    "MedianStopping": MedianStoppingRule,
-    "HyperBand": HyperBandScheduler,
-    "AsyncHyperBand": AsyncHyperBandScheduler,
-}
+def import_schedulers():
+    for key in ["FIFO", "MedianStopping", "HyperBand",
+                "AsyncHyperBand", "HyperOpt"]:
+        try:
+            _SCHEDULERS[key] = get_scheduler(key)
+        except ImportError as e:
+            print("Could not import scheduler: {}".format(key))
```

Comment:
If this is just for HyperOpt, let's just have that fail in the constructor instead of on import.